Dockerfile

```diff
@@ -13,32 +45,33 @@ COPY go.sum go.sum
 # we cache the vendored dependencies before building and
 # copying source so that we don't need to re-download when
 # source changes don't invalidate our downloaded layer
-RUN GO111MODULE=on go mod download
-RUN GO111MODULE=on go mod vendor
+RUN go mod download
+RUN go mod tidy
 
 # copy build manifests
 COPY Makefile Makefile
 
 # copy source files
-COPY cmd/ ./cmd/
-COPY k8s/ ./k8s/
-COPY types/ ./types/
-COPY util/ ./util/
-COPY controller/ ./controller/
+COPY cmd/ cmd/
+COPY k8s/ k8s/
+COPY types/ types/
+COPY util/ util/
+COPY controller/ controller/
 
-# build cspauto binary
-RUN make
+# build cstorpoolauto binary
+RUN make cstorpoolauto
 
-# Use debian as minimal base image to package the final binary
-FROM debian:stretch-slim
+# ---------------------------
+# Use distroless as minimal base image to package the final binary
+# Refer https://github.com/GoogleContainerTools/distroless
+# ---------------------------
+FROM gcr.io/distroless/static:nonroot
```

Comment:
can we just rely on logs?

Reply:
We can try that but sometimes we do require maybe worst case scenario, how do we tackle that.
But I think it would be good to make our logs reliable.

Reply:
I will come back if I face issues.